doc/ref/modules/index.rst

```diff
@@ -376,6 +376,22 @@ The above example will force the minion to use the :py:mod:`systemd
 
 .. __: https://github.com/saltstack/salt/issues/new
 
+Logging Restrictions
+--------------------
+
+As a rule, logging should not be done anywhere in a file before a module is
+loaded. This rule apples to all code that would run before the ``__virtual__()``
+function, as well as the code within the ``__virtual__()`` function itself.
+
+If logging statements are called before the ``__virtual__()`` function determines
```

Comment:
This sentence seems like it's missing a word? Before...what? The function itself?